lib/searchkick/bulk_indexer.rb

```diff
@@ -56,27 +56,24 @@ def batches_left
     end
 
     private
-
     def import_or_update(records, method_name, async)
-      if records.any?
-        if async
-          Searchkick::BulkReindexJob.perform_later(
-            class_name: records.first.class.name,
-            record_ids: records.map(&:id),
-            index_name: index.name,
-            method_name: method_name ? method_name.to_s : nil
-          )
-        else
-          records = records.select(&:should_index?)
-          if records.any?
-            with_retries do
-              # call out to index for ActiveSupport notifications
-              if method_name
-                index.bulk_update(records, method_name)
-              else
-                index.bulk_index(records)
-              end
-            end
+      return if records.none?
```

Comment:
This will cause an additional query. I think a better approach would be to call `records.to_a` or `pluck(:id)` first, then check if `any?`.

Reply:
`.to_a` will load all the AR objects into memory that may cause a bloat if we're talking on a considerable amount of AR objects so IMO I think it's best to avoid that.

Although not so pretty I think that the best option here is something like:

```ruby
if async
  record_ids = records.pluck("#{records.klass.table_name}.id")
  return if record_ids.empty?
  Searchkick::BulkReindexJob.perform_later(
    class_name: records.klass.name,
    record_ids: records.pluck(:id),
    index_name: index.name,
    method_name: method_name ? method_name.to_s : nil
  )
else
  loaded_records = records.to_a.select(&:should_index?)
  return if loaded_records.size.zero?
  # ...
end
```

Reply:
That's exactly what I mean above. There's an easy to do this with a minimal change set.